atc/db/build_in_memory_check.go

```diff
@@ -416,10 +419,6 @@ func (b *inMemoryCheckBuild) AcquireTrackingLock(logger lager.Logger, interval t
 }
 
 func (b *inMemoryCheckBuild) Finish(status BuildStatus) error {
-	if !b.runningInContainer {
```

Comment:
@evanchaoli I think your concern is valid so I have updated it to only init the DB while it indeed need to record the build event.

The only thing here is `runningInContainer`, while an error like var source interpolation happens, that value here is false (since the build hasn't really being ran in a container yet). So I have to remove this check to allow it to init the DB.

Reply:
If you remove `runningInContainer`, then all non-run checks will also write to db, which still doesn't resolve my concern. I think you can do something like:

```golang
if !b.runningInContaienr && status == SUCCEED
```

Reply:
In engine.go it only calls b.finish when there is error that needs to be recorded. Why `all non-run checks` would call finish since they haven't started to run yet? Is there somewhere else that method is used?

Reply:
Also, `Finish` clean up entries in containers and resource caches table, if we gate it by `succeeded`, what will happen to the containers and caches used for the succeeded check build?

Reply:
We don't simply gate it by `succeeded`. When 'runningInContainer` is true, it's not gated, so clean up will still run.

Reply:
I see. So if `runningInContainer` is false and it is succeeded, it means the check build finds a version so it doesn't need to create a container to run the check, and thus it won't need to run clean up in `Finish`.